# Week 2 - Build a classification Model
**Vortex Tech AI & ML Internship — Week 2**

Predict whether a person has heart disease using scikit-learn.

**Dataset-** [Heart disease dataset](ttps://www.kaggle.com/datasets/pratyushpuri/heart-disease-dataset-3k-rows-python-code-2025)

## Step 1: Import libraries 

In [11]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.tree import DecisionTreeClassifier
from sklearn.metrics import accuracy_score, precision_score,recall_score,f1_score
# This are the libraries that we will use for our machine learning model.

## Step 2: Load data

In [12]:
df = pd.read_csv('heart_disease_dataset.csv')
print(df.shape)
df.head()

(3069, 17)


,age,sex,cp,trestbps,chol,fbs,restecg,thalach,exang,oldpeak,slope,ca,thal,smoking,diabetes,bmi,heart_disease
0,67,1,2,111,536,0,2,88,0,1.3,3,2,3,1,0,23.4,1
1,57,1,3,109,107,0,2,119,0,5.4,2,0,3,0,1,35.4,0
2,43,1,4,171,508,0,1,113,0,3.7,3,0,7,1,1,29.9,0
3,71,0,4,90,523,0,2,152,0,4.7,2,1,3,1,0,15.2,1
4,36,1,2,119,131,0,2,128,0,5.9,3,1,3,1,0,16.7,1


In [13]:
print(df.columns.tolist())
df.info()

['age', 'sex', 'cp', 'trestbps', 'chol', 'fbs', 'restecg', 'thalach', 'exang', 'oldpeak', 'slope', 'ca', 'thal', 'smoking', 'diabetes', 'bmi', 'heart_disease']


<class 'pandas.DataFrame'>
RangeIndex: 3069 entries, 0 to 3068
Data columns (total 17 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   age            3069 non-null   int64  
 1   sex            3069 non-null   int64  
 2   cp             3069 non-null   int64  
 3   trestbps       3069 non-null   int64  
 4   chol           3069 non-null   int64  
 5   fbs            3069 non-null   int64  
 6   restecg        3069 non-null   int64  
 7   thalach        3069 non-null   int64  
 8   exang          3069 non-null   int64  
 9   oldpeak        3069 non-null   float64
 10  slope          3069 non-null   int64  
 11  ca             3069 non-null   int64  
 12  thal           3069 non-null   int64  
 13  smoking        3069 non-null   int64  
 14  diabetes       3069 non-null   int64  
 15  bmi            3069 non-null   float64
 16  heart_disease  3069 non-null   int64  
dtypes: float64(2), int64(15)
memory usage: 407.7 KB


## Step 3: Identify target and feature columns

**Target** = ` heart_disease ` (0 = no disease, 1 = disease) 

**Features** = Every other column.

In [14]:
TARGET_COL = 'heart_disease'

df = df.dropna(subset = [TARGET_COL])

X = df.drop(columns= [TARGET_COL])
y = df[TARGET_COL]

y.value_counts()

heart_disease
0    1878
1    1191
Name: count, dtype: int64

## Step 4: Train/Test Split 


- The ratio of the split is 80% for training and 20% for testing.
- The random state is set to 42 to ensure reproducibility of the results.

In [15]:
X_train, X_test, y_train, y_test = train_test_split(X,y,test_size = 0.2,random_state = 42)
 
print(X_train.shape,X_test.shape)

(2455, 16) (614, 16)


## Step 5: Train Logistic Regression

In [16]:
model = LogisticRegression(max_iter = 2000)
model.fit(X_train,y_train)
predictions = model.predict(X_test)

Evaluating the model using accuracy.precision,recall and f1 score metrics.

In [17]:

acc = accuracy_score(y_test,predictions)
pre = precision_score(y_test,predictions)
rec = recall_score(y_test,predictions)
f1=  f1_score(y_test,predictions)

print(f"Accuracy : {acc:.3f}")
print(f"Precision : {pre:.3f}")
print(f"Recall : {rec:.3f}")
print(f"F1 score : {f1:.3f}")

Accuracy : 0.650
Precision : 0.590
Recall : 0.400
F1 score : 0.477


## Step 6:  Decision Tree Comparison
Comparing the performance of decision tree classifier with logistic regression model.

In [18]:

tree_model = DecisionTreeClassifier(random_state = 42)
tree_model.fit(X_train,y_train)
tree_pred = tree_model.predict(X_test)

print(f"Accuracy: {accuracy_score(y_test,tree_pred):.3f}")
print(f"Precision: {precision_score(y_test,tree_pred):.3f}")
print(f"Recall: {recall_score(y_test,tree_pred):.3f}")
print(f"F1 Score: {f1_score(y_test,tree_pred):.3f}")

Accuracy: 0.593
Precision: 0.490
Recall: 0.486
F1 Score: 0.488


## MODEL PERFORMANCE SUMMARY
In this notebook, i have built two `machine learning` models to predict *heart disease*. The first model is a logistic regression model and the second model is a decision tree classifier.

**The Logistic Regression model achieved**
- **Accuracy** : 0.650
- **Precision** : 0.590
- **Recall** : 0.400
- **F1 Score** : 0.477

**The Decision Tree Classifier model achieved**
- **Accuracy** : 0.593
- **Precision** : 0.490
- **Recall** : 0.486
- **F1 Score** : 0.488

We can see that the `Logistic Regression` model performed better than the `Decision Tree Classifier` model in terms of accuracy, precision, and F1 score. However, the Decision Tree Classifier model had a slightly better recall score. Overall, the Logistic Regression model is a better choice for predicting heart disease in this dataset.